fw/http_msg.c

```diff
@@ -815,6 +820,47 @@ __hdr_sub(TfwHttpMsg *hm, const TfwStr *hdr, unsigned int hid)
 	return TFW_PASS;
 }
 
+/**
+ * Substitute request method
+ *
+ * For a while it's just a custom realization for substituting dest method with sub method
+ * (Implemented to substitute PURGE with GET in X-Tempesta-Cache processing)
+ */
+int
+tfw_http_req_sub(TfwHttpMsg *hm, unsigned char dest_meth, unsigned char sub_meth)
+{
+	const char *dest_name = NULL;
+	const char *sub_name = NULL;
+
+	size_t dest_len, sub_len;
+
+	TfwStr sub_str = {};
+
+	if (hm->req->method != dest_meth)
+		return TFW_BLOCK; /* todo: check return values */
+
+	/* do we need to look for "get" string in HTTP headers or just set it as stub constant value? */
+	dest_name = tfw_http_tbl_method_get_name(dest_meth);
+	sub_name = tfw_http_tbl_method_get_name(sub_meth);
+
+	/* todo: optimize this later */
+	if (!dest_name || !sub_name)
+		return TFW_BLOCK;
+
+	dest_len = strlen(dest_name);
+	sub_len = strlen(sub_name);
```

Comment:
We do our best to make as many things computed statically as possible, so we call functions like `strlen()` in very rare cases. You basically don't need the lookup logic and length calculation if you reduce the function to handle substitution to "GET" only.